lib/Doctrine/DBAL/Schema/AbstractAsset.php

```diff
@@ -172,13 +152,14 @@ public function getName()
     /**
      * Gets the quoted representation of this asset but only if it was defined with one. Otherwise
      * return the plain unquoted value as inserted.
-     *
-     * @return string
      */
-    public function getQuotedName(AbstractPlatform $platform)
+    public function getQuotedName(AbstractPlatform $platform) : string
     {
+        $name = $this->getName();
+        assert(is_string($name));
```

Comment:
The assertion will fail if the method is called in an anonymous asset. Should be an exception instead.